Google.Api.Gax.Grpc/Rest/HttpRulePathPattern.cs

```diff
@@ -80,15 +92,27 @@ internal static HttpRulePathPattern Parse(string pattern, MessageDescriptor requ
             {
                 return message => (string) accessor.GetValue(message);
             }
-            else
+
+            var remainder = CreatePropertyAccessor(field.MessageType, propertyPath.Substring(periodIndex + 1));
+            return message =>
             {
-                var remainder = CreatePropertyAccessor(field.MessageType, propertyPath.Substring(periodIndex + 1));
-                return message =>
-                {
-                    var nested = (IMessage)accessor.GetValue(message);
-                    return nested is null ? "" : remainder(nested);
-                };
-            }
+                var nested = (IMessage)accessor.GetValue(message);
+                return nested is null ? "" : remainder(nested);
+            };
+        }
+
+        internal sealed class HttpRulePathPatternSegment
+        {
+            private readonly Func<IMessage, string> _formatter;
+            internal string TopLevelFieldName { get; }
```

Comment:
I think a comment would be useful here. I'm not really following how this is expected to be used.